### ISSUE TITLE
Functions That Search JSON Values

### ISSUE THREAD

**Issue by dengliu** [StarRocks/starrocks#10556]
## Functions That Search JSON Values

**Is your feature request related to a problem? Please describe.**
We are load json type data to SRs. Would be great to see SRs support more functions that search JSON Values.
E.g. `JSON_KEYS, JSON_SEARCH, JSON_CONTAINS, JSON_VALUE`, etc

**Describe the solution you'd like**
Be compatible with the JSON functions in MySQL
https://dev.mysql.com/doc/refman/8.0/en/json-search-functions.html

**Describe alternatives you've considered**
N/A
**Additional context**
N/A

**Comment by mofeiatwork** [comment]
1. `JSON_VALUE` is almost equivalent to our `JSON_QUERY`, which could search a specified key of JSON document.
2. `JSON_KEYS` could be implemented at 2.4
3. `JSON_SEARCH` is pretty complex, but could be replaced with `JSON_QUERY` and string functions as a workaround
4. `JSON_CONTAINS` is almost equivalent to our `JSON_EXISTS`, which could check specified JSON path

**Comment by jaogoy** [comment]
@mofeiatwork But, JSON_VALUE returns SQL type directly, it's more convenionent than JSON_QUERY to extract data of primary type, while you should cast the data to primary type by yourself when using JSON_QUERY.

**Comment by dengliu** [comment]
Thanks all!
As more and more data, logs, events are coming in JSON format. 
It would be great to see better JSON data support in SRs. 

The list of JSON functions in mysql are pretty handy:
https://dev.mysql.com/doc/refman/8.0/en/json-search-functions.html

**Comment by dengliu** [comment]
> @mofeiatwork But, JSON_VALUE returns SQL type directly, it's more convenionent than JSON_QUERY to extract data of primary type, while you should cast the data to primary type by yourself when using JSON_QUERY.

correct. In a recent query example, the object value is an INT in json, but I have to cast it twice in SRs SQL to get the right value.
```
cast(cast(value -> 'row_count' as STRING) as BIGINT)
```
if we have `JSON_VALUE` we can simply do this `JSON_VALUE(value -> 'row_count' , RETURNING BIGINT)`

**Comment by mofeiatwork** [comment]
> > @mofeiatwork But, JSON_VALUE returns SQL type directly, it's more convenionent than JSON_QUERY to extract data of primary type, while you should cast the data to primary type by yourself when using JSON_QUERY.
> 
> correct. In a recent query example, the object value is an INT in json, but I have to cast it twice in SRs SQL to get the right value.
> 
> ```
> cast(cast(value -> 'row_count' as STRING) as BIGINT)
> ```
> 
> if we have `JSON_VALUE` we can simply do this `JSON_VALUE(value -> 'row_count' , RETURNING BIGINT)`

Got it. We would optimize the type casting of JSON recently.
